flink-ml-core/src/main/java/org/apache/flink/ml/linalg/typeinfo/DenseMatrixSerializer.java

```diff
@@ -0,0 +1,138 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.flink.ml.linalg.typeinfo;
+
+import org.apache.flink.api.common.typeutils.SimpleTypeSerializerSnapshot;
+import org.apache.flink.api.common.typeutils.TypeSerializerSnapshot;
+import org.apache.flink.api.common.typeutils.base.TypeSerializerSingleton;
+import org.apache.flink.core.memory.DataInputView;
+import org.apache.flink.core.memory.DataOutputView;
+import org.apache.flink.ml.linalg.DenseMatrix;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+/** Specialized serializer for {@code DenseMatrix}. */
+public final class DenseMatrixSerializer extends TypeSerializerSingleton<DenseMatrix> {
+
+    private static final long serialVersionUID = 1L;
+
+    private static final double[] EMPTY = new double[0];
+
+    private static final DenseMatrixSerializer INSTANCE = new DenseMatrixSerializer();
+
+    @Override
+    public boolean isImmutableType() {
+        return false;
+    }
+
+    @Override
+    public DenseMatrix createInstance() {
+        return new DenseMatrix(0, 0, EMPTY);
+    }
+
+    @Override
+    public DenseMatrix copy(DenseMatrix from) {
+        return new DenseMatrix(
+                from.numRows, from.numCols, Arrays.copyOf(from.values, from.values.length));
+    }
+
+    @Override
+    public DenseMatrix copy(DenseMatrix from, DenseMatrix reuse) {
+        if (from.values.length == reuse.values.length) {
+            System.arraycopy(from.values, 0, reuse.values, 0, from.values.length);
+            return reuse;
+        }
+        return copy(from);
+    }
+
+    @Override
+    public int getLength() {
+        return -1;
+    }
+
+    @Override
+    public void serialize(DenseMatrix matrix, DataOutputView target) throws IOException {
+        if (matrix == null) {
+            throw new IllegalArgumentException("The matrix must not be null.");
+        }
+
+        final int len = matrix.values.length;
+        target.writeInt(matrix.numRows);
+        target.writeInt(matrix.numCols);
+        for (int i = 0; i < len; i++) {
+            target.writeDouble(matrix.values[i]);
+        }
+    }
+
+    @Override
+    public DenseMatrix deserialize(DataInputView source) throws IOException {
+        int m = source.readInt();
+        int n = source.readInt();
+        double[] values = new double[m * n];
+        for (int i = 0; i < m * n; i++) {
+            values[i] = source.readDouble();
+        }
+        return new DenseMatrix(m, n, values);
+    }
+
+    private static void readDoubleArray(double[] dst, DataInputView source, int len)
```

Comment:
Can we make the coding style consistent?
For example, can we also use `readDoubleArray` in `public DenseMatrix deserialize(DataInputView source)` or just remove this method?